src/Shell/Task/TestTask.php

```diff
@@ -513,6 +515,10 @@ public function generateConstructor($type, $fullClassName)
             $pre = "\$this->io = \$this->getMock('Cake\Console\ConsoleIo');";
             $construct = "new {$className}(\$this->io);";
         }
+        if ($type === 'task') {
+            $pre = "\$this->io = \$this->getMock('Cake\Console\ConsoleIo');\n";
+            $construct = "\$this->getMock('{$fullClassName}', ['in', 'out', 'err', '_stop'], [\$this->io]);";
```

Comment:
We should fix up those test cases then. Before `ConsoleIo` existed it was very necessary to mock all those methods.